File: src/main/java/net/theevilreaper/vulpes/backend/controller/ItemController.java
package net.theevilreaper.vulpes.backend.controller;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.*;
import io.micronaut.http.annotation.Controller;
import net.theevilreaper.vulpes.api.model.ItemModel;
import net.theevilreaper.vulpes.api.repository.ItemRepository;
import jakarta.inject.Inject;

import java.util.List;

/**
 * @author theEvilReaper
 * @version 1.0.0
 * @since 1.0.0
 */
@Controller("/item")
public class ItemController {

    private final ItemRepository itemRepository;

    @Inject
    public ItemController(ItemRepository itemRepository) {
        this.itemRepository = itemRepository;
    }

    @Post
    @Produces(MediaType.APPLICATION_JSON)
    public HttpResponse<ItemModel> add(@Body ItemModel item) {
        return itemRepository.add(item);
    }

    @Get("/{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public HttpResponse<ItemModel> getById(@PathVariable String id) {
        return itemRepository.getByID(id);
    }

    @Delete("/remove/{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public HttpResponse<ItemModel> remove(@PathVariable String id) {
        return itemRepository.delete(id);
    }

    @Get("/getAll")
    @Produces(MediaType.APPLICATION_JSON)
    public HttpResponse<List<ItemModel>> getAll() {
        return itemRepository.getAll();
    }

    @Delete("/deleteAll")
    @Produces(MediaType.APPLICATION_JSON)
    public HttpResponse<List<ItemModel>> deleteAll() {
        return itemRepository.deleteAll();
    }

    @Post("/update")
    @Produces(MediaType.APPLICATION_JSON)
    public HttpResponse<ItemModel> update(@Body ItemModel model) {
        return itemRepository.update(model);
    }
}

File: src/main/kotlin/net/theevilreaper/vulpes/backend/controller/ItemController.kt
package net.theevilreaper.vulpes.backend.controller

import net.theevilreaper.vulpes.api.model.ItemModel
import net.theevilreaper.vulpes.backend.dao.DatabaseAccessObject
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

/**
 * @author theEvilReaper
 * @version 1.0.0
 * @since
 **/

@CrossOrigin(
    origins = ["*"],
    maxAge = 4800,
    allowCredentials = "false",
    methods = [RequestMethod.POST, RequestMethod.DELETE, RequestMethod.GET, RequestMethod.HEAD, RequestMethod.OPTIONS]
)
@RestController
class ItemController(
    private val itemDatabaseHandler: DatabaseAccessObject<ItemModel>
) {
    @RequestMapping("/item", method = [RequestMethod.POST], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun add(@RequestBody item: ItemModel): ResponseEntity<ItemModel> {
        // Validation
        return itemDatabaseHandler.add(item)
    }

    @RequestMapping("/item/{id}", method = [RequestMethod.GET], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun getByID(@PathVariable("id") id: String): ResponseEntity<ItemModel> {
        return itemDatabaseHandler.getByID(id)
    }

    @RequestMapping("/item/remove/{id}", method = [RequestMethod.DELETE], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun remove(@PathVariable("id") id: String): ResponseEntity<ItemModel> {
        return itemDatabaseHandler.delete(id)
    }

    @RequestMapping("/item/getAll", method = [RequestMethod.GET], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun getAll(): ResponseEntity<List<ItemModel>> {
        return itemDatabaseHandler.getAll()
    }

    @DeleteMapping("/item/deleteAll")
    fun deleteAll(): ResponseEntity<List<ItemModel>> {
        return itemDatabaseHandler.deleteAll()
    }

    @RequestMapping("/item/update", method = [RequestMethod.POST], produces = [MediaType.APPLICATION_JSON_VALUE])
    fun update(@RequestBody model: ItemModel): ResponseEntity<ItemModel> {
        return itemDatabaseHandler.update(model)
    }
}
